ZUGFeRD/InvoiceDescriptor1Writer.cs

```diff
@@ -360,17 +358,47 @@ public override void Save(InvoiceDescriptor descriptor, Stream stream, ZUGFeRDFo
                 Writer.WriteEndElement();
             }
 
-            if (this.Descriptor.PaymentTerms != null)
+            //  The cardinality depends on the profile.
+            switch (Descriptor.Profile)
             {
-                Writer.WriteStartElement("ram:SpecifiedTradePaymentTerms");
-                Writer.WriteOptionalElementString("ram:Description", this.Descriptor.PaymentTerms.Description);
-                if (this.Descriptor.PaymentTerms.DueDate.HasValue)
-                {
-                    Writer.WriteStartElement("ram:DueDateDateTime");
-                    _writeElementWithAttribute(Writer, "udt:DateTimeString", "format", "102", _formatDate(this.Descriptor.PaymentTerms.DueDate.Value));
-                    Writer.WriteEndElement(); // !ram:DueDateDateTime
-                }
-                Writer.WriteEndElement();
+                case Profile.Unknown:
+                case Profile.Minimum:
+                    break;
+                case Profile.Extended:
+                    foreach (PaymentTerms paymentTerms in this.Descriptor.PaymentTerms)
+                    {
+                        Writer.WriteStartElement("ram:SpecifiedTradePaymentTerms");
+                        Writer.WriteOptionalElementString("ram:Description", paymentTerms.Description);
+                        if (paymentTerms.DueDate.HasValue)
+                        {
+                            Writer.WriteStartElement("ram:DueDateDateTime");
+                            _writeElementWithAttribute(Writer, "udt:DateTimeString", "format", "102", _formatDate(paymentTerms.DueDate.Value));
+                            Writer.WriteEndElement(); // !ram:DueDateDateTime
+                        }
+                        Writer.WriteEndElement();
+                    }
+                    break;
+                default:
+                    if (Descriptor.PaymentTerms.Count > 0)
+                    {
+                        Writer.WriteStartElement("ram:SpecifiedTradePaymentTerms");
+                        var sbPaymentNotes = new StringBuilder();
+                        DateTime? dueDate = null;
+                        foreach (PaymentTerms paymentTerms in this.Descriptor.PaymentTerms)
+                        {
+                            sbPaymentNotes.AppendLine(paymentTerms.Description);
+                            dueDate = dueDate ?? paymentTerms.DueDate;

```

Comment:
uh, that looks a bit risky (same for all versions). What if the due dates are different between the different entries? In this case, we should imho rather throw an exception instead of "manipulating" the output. What do you think?

Reply:
It was currently an (obvious) choice, if my understanding of the single use of the date is correct regarding the profiles using single cardinality.

The approach is to use the first one found in multiple PaymentTerms supplied and concatenate the descriptions.

Other suggestions are most welcome.

Reply:
@stephanstapel any ideas on how to handle this differently?
Current outcome based on single `PaymentTerms` is still identical to the previous implementation.
Given a cardinality of [0..1] in profiles other than Extended the code uses the due date of the first (and probably only) payment terms, and writes them to the XML. I was considering to use either a `SortedList` or even a `Queue` but that would have added way more complexity.

Right now, there would be an unexpected scenario if the first payment terms _do not have a due date_ but any following one. Question is how likely this constellation is? Plus, it could be easily mitigated by the package consumer specifying a due date with the first payment terms (which is highly plausible).

What's your take on this?

Ohh, fun fact, the sequence of XML elements written is important!